packages/yew-router/src/router.rs

```diff
@@ -72,16 +74,37 @@ fn base_router(props: &RouterProps) -> Html {
         basename,
     } = props.clone();
 
+    let basename = basename.map(|m| strip_slash_suffix(&m).to_string());
+    let navigator = Navigator::new(history.clone(), basename.clone());
+
+    let old_basename = use_state_eq(|| Option::<String>::None);
+    if basename != *old_basename {
+        // If `old_basename` is `Some`, path is probably prefixed with `old_basename`.
+        // If `old_basename` is `None`, path may or may not be prefixed with the new `basename`,
+        // depending on whether this is the first render.
+        let old_navigator = Navigator::new(
+            history.clone(),
+            old_basename.as_ref().or(basename.as_ref()).cloned(),
+        );
+        old_basename.set(basename.clone());
+        let location = history.location();
+        let stripped = old_navigator.strip_basename(Cow::from(location.path()));
+        let prefixed = navigator.prefix_basename(&stripped);
+
+        if prefixed != location.path() {
+            history
+                .replace_with_query(prefixed, Raw(location.query_str()))
+                .unwrap_or_else(|never| match never {});
+        }
+    }
```

Comment:
Are you sure this doesn't need to be in a `use_effect`? Setting a `use_state` from the function component body is wrong here. It can trigger an infinite loop. See this [SO question](https://stackoverflow.com/questions/74215238/why-react-goes-infinite-when-i-set-state-in-function-body) about react, but is still relevant here

Reply:
I don't see how `use_effect` is necessary. For one thing, `use_effect` (because it is a hook) can't be conditional. I would have to move the conditional into the `use_effect`. I didn't want to delay the `replace_with_query` until after the first render. Do you think this would help?
```rust
    let mut new_basename = None; // NEW

    if basename != *old_basename {
        // If `old_basename` is `Some`, path is probably prefixed with `old_basename`.
        // If `old_basename` is `None`, path may or may not be prefixed with the new `basename`,
        // depending on whether this is the first render.
        let old_navigator = Navigator::new(
            history.clone(),
            old_basename.as_ref().or(basename.as_ref()).cloned(),
        );
        new_basename = Some(basename.clone());                                // CHANGED
        let location = history.location();
        let stripped = old_navigator.strip_basename(Cow::from(location.path()));
        let prefixed = navigator.prefix_basename(&stripped);

        if prefixed != location.path() {
            history
                .replace_with_query(prefixed, Raw(location.query_str()))
                .unwrap_or_else(|never| match never {});
        }
    }
    use_effect(move || {                                      // NEW
        if let Some(new_basename) = new_basename {
            basename.set(new_basename);
        }
    });
```

Reply:
But perhaps I should use `use_mut_ref` instead of `use_state`, since I don't think the rerender is useful..